User Profiles

elbotho opened the pull request:
rebase of #573

### example profile
https://frontend-git-profiles-rebase.serlo.vercel.app/user/profile/wolfgang
https://frontend-git-profiles-rebase.serlo.vercel.app/user/profile/kathongi

### issue
closes https://github.com/serlo/frontend/issues/677

### status
- Uses mockup data for now (because descriptions are not part of the staging data (returns NULL)

elbotho committed:
first draft

elbotho committed:
add descriptions

elbotho committed:
update fetcher test for User

elbotho committed:
use new timeago component

elbotho committed:
use as little mockup data as possible

elbotho committed:
new user-link component with badges

elbotho committed:
add bird image to profile

elbotho committed:
add birds and community roles to profiles

elbotho committed:
add birds and community roles to profiles part 2

elbotho committed:
add donation form to own profile

elbotho committed:
donation form responsive

elbotho committed:
translate

elbotho committed:
add user id to donation form

elbotho committed:
cleanup rebase

elbotho committed:
prepare for new features

inyono commented:
FYI: staging has anonymized data (e.g. removes all email addresses & descriptions of users). This would work in prod.

elbotho committed:
add comments to profiles

elbotho committed:
add edit profile link to own profile

elbotho committed:
link to /user/profile/${username} in menu

elbotho committed:
add fallback for legacy routes /user/public and /user/me

elbotho commented:
@Entkenntnis okay! This is ready for review now. 
As per your suggestion I implemented the link to /user/profile/username in the menu (best solution) and added a simple fallback redirect for /user/me and /user/profile that we can remove after 100%.

elbotho committed:
Merge branch 'staging' into profiles-rebase

Entkenntnis commented on src/components/content/author-tools.tsx:
```diff
@@ -30,6 +30,7 @@ export function AuthorTools({ data }: AuthorToolsProps) {
       content={<AuthorToolsHoverMenu data={data} />}
     >
       <EditButton>
+        author tools
```
Hm, looks superfluous here

Entkenntnis commented on src/components/pages/user/profile-community-banner.tsx:
```diff
@@ -0,0 +1,180 @@
+  ) {
+    if (isOwnProfile) {
+      const counts = fetchGroupsTotalCounts()
+      const parts = bannerContent[key].ownProfileMessage.split('%no%')
```
We have something better now :)

Entkenntnis commented on src/components/pages/user/profile-community-banner.tsx:
```diff
@@ -0,0 +1,180 @@
+
+    return (
+      <StyledP>
+        {bannerContent[key].otherUserProfileMessage.replace(
```
Same here :)

Entkenntnis reviewed (changes requested):
Looks fine! (Finally getting this done)

elbotho committed:
use new translation function

elbotho commented:
Thanks @Entkenntnis
